highfidelity/hifi
# FB 4156; Fix GoTo text field focus; other GoTo tweaks

Fixes:
- Forward/back button icon y-alignment
- Forward/back button hover/click visuals and behavior
- Text field auto-focus in Desktop mode
- GoTo show/hide/toggle behavior fixed
- Fixed warnings in console
- Fixed visual bugs around text field in desktop mode
- Added padding to text input field to be more in-line with UIT
- Cleaned up help text

*Test Plan:*
Precondition: Start up in home (so that there are no other places in your session history). Start up in Desktop mode.

1. In Desktop mode, open the GoTo address bar.
a. The input field should have keyboard focus: normal cursor (if any).
b. The input field should tell you where you are.
2. Enter some text using your physical keyboard.
a. The gray ""Your Location"" text should go away and your text should appear.
b. As you type each letter (or the delete key), the character should appear in the text field (or the previous letter should disappear if deleting).
c. As you type each letter (or the delete key), the suggestions should be re-filtered (on each key).
3. Click with the mouse off of the address bar (with text still entered)
a. The input field should lose focus.
4. Switch to HMD. In HMD, tap with the stylus/finger on the text input field. Then, input some text using the virtual keyboard
a. The input field should have keyboard focus: normal cursor
b. The virtual keyboard should appear, and the virtual keyboard should be the same size as elsewhere in the application.
c. Same results as in (2).
7. Switch to Desktop. Click the home button in the GoTo address bar.
a. The address bar should close.
b. The GoTo button should become unhighlighted.
8. Open the GoTo address bar.
a. The back and forward buttons should be disabled - visually and in fact.
9. Go someplace else (i.e. dev-welcome). Re-open the GoTo menu.
a. The back button should be enabled visually.
10. Click the back button
a. You should be back home
b. The location field should reflect that you are home.
c. The forward button should be enabled visually.
11. Click the forward button.
a. You should be at the same place as in (9)
b. The location field should reflect that.
